core/templates/dev/head/domain/skill/skill-rights-backend-api.service.ts

```diff
@@ -16,92 +16,97 @@
-       * behavior and returning a promise object), except this function will
-       * attempt to see whether the given skill rights has been
-       * cached. If it has not yet been cached, it will fetch the skill
-       * rights from the backend. If it successfully retrieves the skill
-       * rights from the backend, it will store it in the cache to avoid
-       * requests from the backend in further function calls.
-       */
-      loadSkillRights: function(skillId) {
-        return $q(function(resolve, reject) {
-          if (_isCached(skillId)) {
+  _isCached(skillId: string) {
+    return this.skillRightsCache.hasOwnProperty(skillId);
+  }
+  /**
+    * Gets a skill's rights, given its ID.
+    */
+  fetchSkillRights(skillId: string) {
+    return new Promise((resolve, reject) => {
+      this._fetchSkillRights(skillId, resolve, reject);
+    });
+  }
+  /**
+    * Behaves exactly as fetchSkillRights (including callback
+    * behavior and returning a promise object), except this function will
+    * attempt to see whether the given skill rights has been
+    * cached. If it has not yet been cached, it will fetch the skill
+    * rights from the backend. If it successfully retrieves the skill
+    * rights from the backend, it will store it in the cache to avoid
+    * requests from the backend in further function calls.
+    */
+  loadSkillRights(skillId: string) {
+    return new Promise((resolve, reject) => {
+      if (this._isCached(skillId)) {
+        if (resolve) {
+          resolve(this.skillRightsCache[skillId]);
+        }
+      } else {
+        this._fetchSkillRights(skillId,
+          (skillRights: ISkillRightBackendInterface) => {
+            this.cacheSkillRights(skillId, skillRights);
             if (resolve) {
-              resolve(skillRightsCache[skillId]);
+              resolve(this.skillRightsCache[skillId]);
             }
-          } else {
-            _fetchSkillRights(skillId, function(skillRights) {
-              skillRightsCache[skillId] = skillRights;
-              if (resolve) {
-                resolve(skillRightsCache[skillId]);
-              }
-            }, reject);
-          }
-        });
-      },
+          }, reject);
+      }
+    });
+  }
 
-      /**
-       * Returns whether the given skill rights is stored within the
-       * local data cache or if it needs to be retrieved from the backend
-       * upon a laod.
-       */
-      isCached: function(skillId) {
-        return _isCached(skillId);
-      },
+  isCached(skillId: string) :boolean {
+    return this._isCached(skillId);
+  }
 
-      /**
-       * Replaces the current skill rights in the cache given by the
-       * specified skill ID with a new skill rights object.
-       */
-      cacheSkillRights: function(skillId, skillRights) {
-        skillRightsCache[skillId] = angular.copy(skillRights);
-      }
-    };
+  cacheSkillRights(skillId: string,
+      skillRights: ISkillRightBackendInterface): void{
```

Comment:
Add space after void.

+cc @oppia/dev-workflow-team Shouldn't the linter catch this?

Reply:
Hi @Showtim3 -- I think the linter team has a bit of a backlog. Could you please add this to the list in #8423 and link back to this comment from there?

Reply:
Hey @seanlip, I have added it to the list. Let me know if there's something else to be done.

Reply:
Thanks! All good, thanks for the update.